### DISCUSSION TITLE
Please allow approval of specific files in PRs; not overall approvals

### DISCUSSION THREAD

**Discussion by Keith-McKinley** [community/community#81192] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

We are enjoying the Code Owners feature in our work in GitHub, but have noticed two shortcomings that would make the feature far more valuable:

- It is hard to see in a PR the files for which you are a code owner. Our technical leaders often need to review a PR only because of one or two files, and it would help tremendously if the PR UI made it easy for them to readily notice what changes in the PR are requiring them as a codeowner.
- When one of those leaders reviews the files they are required for (as a code owner), they may want to approve the changes, but have not necessarily done a complete and careful review of the entire PR. The repo and branch settings require two approvals, but that tech leader doesn't want to approve the PR because they didn't review all of it. Thus they wait until others do a more complete code review, then add their approval last. This makes for a lot of process hinderance.

Having the PR point out the files which the current user is a code owner for (perhaps putting the list of owners in the file header on the 'Files changed' tab?, then allowing the reviewer to only approve the files which they checkbox as 'viewed' is a good answer. That way once codeowners have approved files they had to review and other restrictions are met, it can be merged

**Comment by cbejensen** [comment]
The first feature is available today:
![image](https://github.com/user-attachments/assets/f4f2c6b0-51ac-4875-9e83-07e2956d07d6)

Would love to see the second as well.
